packages/vx-xychart/src/components/series/LineSeries.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useContext, useCallback, useEffect } from 'react';
+import LinePath from '@vx/shape/lib/shapes/LinePath';
+import { ScaleConfig, ScaleConfigToD3Scale, ScaleInput } from '@vx/scale';
+import { AxisScaleOutput } from '@vx/axis';
+import DataContext, { InferDataContext } from '../../context/DataContext';
+import isValidNumber from '../../typeguards/isValidNumber';
+import { DataContext as DataContextType } from '../../types/data';
+
+type LineSeriesProps<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  YScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  Datum
+> = {
+  dataKey: string;
+  data: Datum[];
+  xAccessor: (d: Datum) => ScaleInput<ScaleConfigToD3Scale<XScaleConfig, AxisScaleOutput>>;
+  yAccessor: (d: Datum) => ScaleInput<ScaleConfigToD3Scale<YScaleConfig, AxisScaleOutput>>;
+};
+
+export default function LineSeries<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  YScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  Datum
+>({
+  data,
+  xAccessor,
+  yAccessor,
+  dataKey,
+  ...lineProps
+}: LineSeriesProps<XScaleConfig, YScaleConfig, Datum>) {
+  const { xScale, yScale, colorScale, dataRegistry } = useContext(DataContext);
+
+  // register data on mount
+  // @TODO(chris) make this easier with HOC
+  useEffect(() => {
+    if (dataRegistry) dataRegistry.registerData({ key: dataKey, data, xAccessor, yAccessor });
+    return () => dataRegistry?.unregisterData(dataKey);
+  }, [dataRegistry, dataKey, data, xAccessor, yAccessor]);
+
+  const getScaledX = useCallback(
+    (d: Datum) => {
+      const x = xScale?.(xAccessor(d));
```

Comment:
@kristw I'm curious if you have any thoughts on a final remaining type issue, which is for the accessors here.

![image](https://user-images.githubusercontent.com/4496521/92158263-7cc58c80-ede0-11ea-94d2-e3bb3ac8ebf1.png)

I'm not sure if the `InferDataContext` is the issue (i.e., it's not working and is falling back to `AnyDataContext`), or if the way the `xAccessor` return type is defined ([here](https://github.com/hshoff/vx/pull/789/files#diff-74b1e99c28f45e74a09146815a7831e7R31)) is problematic: 

```ts
xAccessor: (d: Datum) => ScaleInput<ScaleConfigToD3Scale<XScaleConfig, AxisScaleOutput>>;
```

Reply:
i'm pulling the branch to see locally. types are hard to debug without ide

Reply:
Add `, any, any`

```ts
  xAccessor: (
    d: Datum,
  ) => // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ScaleInput<ScaleConfigToD3Scale<XScaleConfig, AxisScaleOutput, any, any>>;
  yAccessor: (
    d: Datum,
  ) => // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ScaleInput<ScaleConfigToD3Scale<YScaleConfig, AxisScaleOutput, any, any>>;
```

Reply:
There are some funkiness with the `DiscreteInput` and `ThresholdInput` generics which are the last two parameters. The `StringLike` has some issue matching.

Reply:
ooooO yes this fixed it, thanks SO MUCH for the help debugging it! 🙏 ❤️